add DUNE style

jonas-eschle opened the pull request:
### Description

This pull request introduces the following changes:
- Added DUNE plot style obtained from `DUNE/dune_plot_style` from [here](https://github.com/DUNE/dune_plot_style/issues/6#issuecomment-2877717053)

jonas-eschle commented:
I've actually updated the PR to have a DUNE and a DUNE1 style, where DUNE points to DUNE1. As other styles have shown (LHCb, ATLAS,...), styles may get updated with more accurate versions or because of changed collaboration guidelines. So it's probably better to be pre-emptive here and let people directly choose, whether they want to have a fixed or an updated style.

andrzejnovak commented on tests/baseline/test_style_dune.png:
The "Simulation Preliminary" text looks top-aligned with "DUNE". I think we strive for the opposite in all the other styles.

jonas-eschle replied:
I agree, bottom would be better. I completely fail to see why it does that. Do you see it?

andrzejnovak replied:
Hmm, could be in https://github.com/scikit-hep/mplhep/blob/633f1e04a934060a8f977fe7cad136edca029c1f/src/mplhep/dune.py#L49 ?

The alignments are set in the loc_dicts https://github.com/scikit-hep/mplhep/blob/main/src/mplhep/label.py#L78 and maybe sth about that kwarg override messes this up? What's strange is that some of the plots look alright like the dune_logo_colors (which btw I don't see the test creating it)

jonas-eschle replied:
I tried that, and the weird thing is, that LHCb has basically the same but aligns correctly. Alright, I keep digging in...

jonas-eschle commented:
Thanks for spotting the alignement, I've fixed it.
Also, we should probably test the latex versions as well? I couldn't make it work, but well.
I've also added the fail-fast: false. This maybe uses slightly more resources (but the tests are anyway so quick...) in case it fails, but helps pinpoint the issue (i.e. if it's OS related). Please revert if you prefer otherwise, but I don't think with this quick testsuite we need to care about resources.

Ready from my side

andrzejnovak commented:
Thanks! It's great to be including DUNE as well. Changes look good from my end. Good to merge once the conflicts get resolved.
